elastic/kibana
# [Alerting] Remove predefined connectors from rule reference array

Resolves https://github.com/elastic/kibana/issues/109247

## Summary

* Updates reading and writing of rule actions to avoid storing predefined connector IDs in the rule references array. Instead, they will be marked with a `preconfigured:` prefix.
* Adds migration to update existing rules to follow the same format. 

## To Verify

* Run this branch and create a rule with some actions, both with preconfigured and non-preconfigured connectors.
* Use Dev Tools to inspect the rule saved object. The actions array should contain the `actionRef` field. For the preconfigured connector, the `actionRef` should look like `preconfigured:${connector.id}`. You should only see entries in the references array for non-preconfigured connectors.
* Verify that this rule and its actions run successfully.
* Create a rule in a previous release like 7.15, with both preconfigured and non-preconfigured connectors. Switch to this branch in order to apply the migration. Verify that the rule and its actions run successfully after migration.


### Checklist

Delete any items that are not applicable to this PR.
